velox/functions/lib/LikeFunctionUtils.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/lib/LikeFunctionUtils.h"
+
+#include <optional>
+
+#include "velox/expression/EvalCtx.h"
+#include "velox/expression/Expr.h"
+#include "velox/functions/lib/ArrayBuilder.h"
+#include "velox/type/StringView.h"
+
+namespace facebook::velox::functions {
+namespace likeopt {
+
+using ::facebook::velox::exec::EvalCtx;
+using ::facebook::velox::exec::Expr;
+using ::facebook::velox::exec::VectorFunction;
+using ::facebook::velox::exec::VectorFunctionArg;
+
+// Check if the pattern to be matched is a constant string with no _ or %
+bool constPatternCheck(const std::string& pattern) {
+  size_t pIdx = 0;
+  size_t pLen = pattern.size();
+
+  while (pIdx < pLen) {
+    if ((pattern[pIdx] == '%') || (pattern[pIdx] == '_')) {
+      return false;
+    }
+    pIdx++;
+  }
+
+  return true;
+}
+
+// In case the pattern to be matched is a constant string with no _ or %, match
+// the inputString with the pattern using a character by character comparison in
+// O(n) time complexity
+bool constMatch(std::string& inputString, const std::string& pattern) {
+  size_t sIdx = 0;
```

Comment:
Velox uses vector_size_t for indexes. Please change to vector_size_t everywhere in this file.

Reply:
Thanks for the suggestion, vector_size_t is int32_t and this gives a warning since size needs to be an unsigned int, hence I used size_t instead